### ISSUE TITLE
categorical_accuracy top_k_categorical_accuracy return different data types

### ISSUE THREAD

**Issue by alexhernandezgarcia** [keras-team/keras#11969]
While `keras.metrics.categorical_accuracy` returns a vector of binary values, `keras.metrics.top_k_categorical_accuracy` returns a scalar. Since both functions represent similar concepts (accuracy), I believe both should return the same data type. Is there a good reason why this is not the case?
